src/main/java/uk/gov/hmcts/reform/sscs/service/LetterUtils.java

```diff
@@ -197,4 +191,123 @@ public static Stream<Pair<String, YesNo>> buildOtherPartiesForReasonableAdjustme
+            boolean isValidRepresentative = sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getRep().getId());
+            if (op.getValue().hasRepresentative() && isValidRepresentative) {
+                return Optional.of(op.getValue().getRep());
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Name> getOtherPartyName(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOriginalSender().getValue().getCode().contains("otherPartyRep")) {
+            Optional<Representative> representative = getRepresentativeOfOtherParty(sscsCaseData);
+            return representative.isPresent() ? Optional.of(representative.get().getName())
+                    : Optional.empty();
+        } else {
+            return sscsCaseData.getOtherParties().stream()
+                    .filter(op -> sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getId()))
+                    .findFirst()
+                    .map(op -> op.getValue().getName());
+        }
+    }
+
+    public static String getNotificationTypeForActionFurtherEvidence(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        boolean isValidActionForNotification = notificationWrapper.getNotificationType().getEvent().equals(EventType.ACTION_FURTHER_EVIDENCE)
+            || notificationWrapper.getNotificationType().getEvent().equals(EventType.POST_HEARING_REQUEST);
+        if (isValidActionForNotification) {
+            if (isValidForSetAsideRequest(notificationWrapper, subscriptionWithType)) {
+                return "confirmation";
+            } else {
+                return "notice";
+            }
+        }
+        return "";
+    }
+
+    static boolean isValidForSetAsideRequest(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        return isValidAppellantForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                || isValidAppellantRepresentativeForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                || isValidJointPartyForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                || isValidOtherParty(notificationWrapper, subscriptionWithType);
+    }
+
+    static boolean isValidAppellantRepresentativeForSetAsideRequest(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        boolean senderIsRepresentative = notificationWrapper.getNewSscsCaseData().getOriginalSender().getValue().getCode().equalsIgnoreCase("representative");
+
+        return  (senderIsRepresentative && (subscriptionWithType.getParty().getId().equalsIgnoreCase(notificationWrapper.getNewSscsCaseData().getAppeal().getAppellant().getId())
+                || subscriptionWithType.getParty().getId().equalsIgnoreCase(notificationWrapper.getNewSscsCaseData().getAppeal().getRep().getId())));
```

Comment:
This is on the right lines, just needs to be applied to every condition in each if statement. This is so that the return statement is blindingly obvious what it is returning, so makes sense at a glance the purpose of the method. (Please rename my suggested booleans if they aren't sensible).

```suggestion
        boolean senderIsRepresentative = notificationWrapper.getNewSscsCaseData().getOriginalSender().getValue().getCode().equalsIgnoreCase("representative");
        
        boolean subscriptionIsAppellant = subscriptionWithType.getParty().getId().equalsIgnoreCase(notificationWrapper.getNewSscsCaseData().getAppeal().getAppellant().getId());
        
        boolean subscriptionIsRep = subscriptionWithType.getParty().getId().equalsIgnoreCase(notificationWrapper.getNewSscsCaseData().getAppeal().getRep().getId());

        return  senderIsRepresentative && (subscriptionIsAppellant || subscriptionIsRep);
```

Repeat for the rest of the static boolean helpers below.

Reply:
Updated.